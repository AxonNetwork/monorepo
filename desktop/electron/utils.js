const path = require('path')

module.exports.getAppPath = function() {
    try {
        return require('electron').app.getAppPath()
    } catch (err) {
        return require('electron').remote.app.getAppPath()
    }
}

module.exports.getEnv = function() {
    const appPath = module.exports.getAppPath()
    const binariesPath = path.join(appPath, '../desktop/build-resources/binaries')
    const env = Object.assign({}, process.env, { CONSCIENCE_APP_PATH: appPath, CONSCIENCE_BINARIES_PATH: binariesPath })
    env.PATH = binariesPath + ':' + env.PATH + ':/usr/local/bin:/usr/bin:/usr/sbin:/sbin'
    return env
}
