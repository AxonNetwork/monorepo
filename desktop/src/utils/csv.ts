export function parseCSV(contents: string){
	const lines = contents.split("\n")
	const data = lines.map(line => line.split(","))
	return data
}