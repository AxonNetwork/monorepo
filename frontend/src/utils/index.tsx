export const isProduction = process.env.NODE_ENV === 'production'

const exec = require('child_process').exec

export async function execCmd(cmd: string, cwd: string) {
    return new Promise((resolve, reject) => {
        exec(cmd, { encoding: 'utf8', cwd }, (error: Error|null, stdout: string, stderr: string) => {
            if (error) {
                reject(error)
            } else {
                resolve({ stdout, stderr })
            }
        })
    })
}
