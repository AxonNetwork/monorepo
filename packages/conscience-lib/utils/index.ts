export { default as autobind } from './autobind'
export { default as strToColor } from './strToColor'
export { default as schemes } from './codeColorSchemes'
export { default as getLanguage } from './getLanguage'
export { default as checkVisible } from './checkVisible'
export { default as nonCacheImg } from './nonCacheImg'
export { repoPageToString, stringToRepoPage } from './parseRepoPage'
export { orgPageToString, stringToOrgPage } from './parseOrgPage'
export { getConscienceURI } from './uri'
export { removeEmail, extractEmail } from './email'
export { parseCSV } from './csv'
export { isTextFile } from './fileTypes'
export { getLastVerifiedEvent, getLastVerifiedEventCommit, getLastVerifiedEventFile, getFirstVerifiedEvent } from './timeline'
