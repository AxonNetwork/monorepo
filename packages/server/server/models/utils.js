import { dynamo } from '../config/aws'

export function makeID() {
    let text = ''
    const possible = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789'

    for (let i = 0; i < 10; i++) {
        text += possible.charAt(Math.floor(Math.random() * possible.length))
    }

    return text
}

export async function getAll(queryParams) {
    let items = []
    let ExclusiveStartKey

    do {
        const resp = await dynamo.queryAsync({
            ...queryParams,
            ExclusiveStartKey,
        })

        if (resp.Items) {
            items = items.concat(resp.Items)
        }

        ExclusiveStartKey = resp.LastEvaluatedKey
    } while (ExclusiveStartKey)

    return items
}
