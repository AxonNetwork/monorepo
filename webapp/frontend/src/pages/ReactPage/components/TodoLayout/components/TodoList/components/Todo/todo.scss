.todo-completed {
  text-decoration: line-through;
  color: grey;
}

.todo-incompleted {
  text-decoration: none;
}
