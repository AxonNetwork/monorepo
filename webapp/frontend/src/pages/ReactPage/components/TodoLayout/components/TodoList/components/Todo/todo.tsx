import React from 'react';

import { ITodo } from 'services/todos/types';

const styles = require('./todo.scss');

interface ITodoProps extends ITodo {
  onClick(): void;
}

export default class Todo extends React.Component<ITodoProps> {
  public onClick = (e: React.MouseEvent<HTMLAnchorElement>) => {
    e.preventDefault();
    this.props.onClick();
  }

  public render() {
    if (this.props.completed) {
      return (
        <a href='#' className='collection-item waves-effect' onClick={this.onClick} >
          <div className={`truncate ${styles['todo-completed']}`}>
            {this.props.text}
          </div>
        </a>
      );
    } else {
      return (
        <a href='#' className='collection-item waves-effect waves-teal' onClick={this.onClick} >
          <div className={`truncate ${styles['todo-incompleted']}`}>
            {this.props.text}
          </div>
        </a>
      );
    }
  }
}
