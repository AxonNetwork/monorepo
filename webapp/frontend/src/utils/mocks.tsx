// Mock dispatch
export const dispatchMock = () => { return; };

// Mock translation
export const tMock = (k: string | string[]) => k;

// Mock i18n
export const i18nMock: any = {
  addResourceBundle: () => { return; },
};
