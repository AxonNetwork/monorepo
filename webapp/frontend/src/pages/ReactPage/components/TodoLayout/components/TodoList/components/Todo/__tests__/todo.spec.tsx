import React from 'react';
import TestRenderer from 'react-test-renderer';

import { dispatchMock } from 'utils/mocks';
import Todo from '../todo';

describe('Todo', () => {
  it('should renders correctly', () => {
    const todo = {
      id: 'id',
      completed: false,
      text: 'text',
    };
    const renderer = TestRenderer.create(
      <Todo {...todo} onClick={dispatchMock} />,
    );
    expect(renderer).toMatchSnapshot();
    renderer.unmount();
  });
});
