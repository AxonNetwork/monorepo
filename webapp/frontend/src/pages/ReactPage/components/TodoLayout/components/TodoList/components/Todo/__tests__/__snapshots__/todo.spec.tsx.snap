// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo should renders correctly 1`] = `
<a
  className="collection-item waves-effect waves-teal"
  href="#"
  onClick={[Function]}
>
  <div
    className="truncate todo-incompleted"
  >
    text
  </div>
</a>
`;
